### PR TITLE
Add failing test case for class generated by Groovy compiler

### PR DESCRIPTION
**Pull request by tmoschou**
This PR adds a failing test case for Groovy class that is causing Spotbugs to throw `edu.umd.cs.findbugs.ba.CFGBuilderException`. It does not fix the issue. Refer to #2145

* Groovy compilation support is added to `spotbugsTestCases/build.gradle`
* `AbstractIntegrationTest` has been cleaned up / refactored to additionally search for `.class` files generated by the Groovy compiler. At the same time simplifies some fall-back search path logic (e.g. presumably left over code for nonexistent `java9/` directory) and switches use of `java.io.File` over to the `java.nio.file` package instead.

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [ ] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code
  _**NOTE: No fix in this PR**_

### CODE REVIEW

**Review by ThrawnCA — APPROVED** [review]

**Review by KengoTODA — APPROVED** [review]
Thanks for your PR! 
<del>I confirmed that the bug comes from 771f26facba9effee104834562803a5361880494 which assumes the size of field is always 1. I will wait for review from other teammates to merge this fix and your test case.</del> I had misunderstanding about the intention of existing code, I will reconsider my fix.

**Review by KengoTODA — APPROVED** [review]
Added a new fix that treats `dup_x1` and `dup_x2` in a better way.

**Comment by KengoTODA** [comment]
I believe this PR can close https://github.com/spotbugs/spotbugs/issues/2145. This PR also close https://github.com/spotbugs/spotbugs/issues/1386.

**Review by ThrawnCA — APPROVED** [review]

**Pull request merged**
